# Aviation Accident Analysis: Identifying Low-Risk Aircraft Options

### 1. Introduction



Any company's venture into aviation has both exciting opportunities and possible risks. This project supports our company's venture into aviation and aircrafts, by analyzing aircraft models and identifying low risk ones; based on the vast amounts of data available. Coming with this data backed evidence, this project aims to drive confident and informed investment decisions in specifically purchasing low risk aircrafts; which may in turn inform other minor decisions in pilot training, maintenance strategies and much more actionable insights to the **Head of the Aviation Division** and all other stakeholders. This analysis is important not only to us as a company and protecting our investments; but also ensuring utmost safety for our pilots and customers.

### 2. Business Problem Statement




Our company with no prior experience in aviation and no knowledge in aircraft models, wants to leverage data analysist to bring insights and new information to inform the kind of aircrafts to invest in that are the lowest risk. Without this analysis the company risks losses in their investments, and worse yet losses of lives of customers and aircraft personnel. The Head of the Aviation Division will recieve actionable recommendations in this regard to make business decisions.

---

### 3. Objectives


 - Analyzing aircraft models to determine those with less severity
 - Finding safe engine designs and what aircrafts they're on
 - Evaluating aircraft models for resilience in diverse weather conditions
 - Determining low risk flight purposes and aircraft models suited.

### 4. Data Understanding


We'll start by breaking down our dataset to understand the data leading our analysis.


#### 4.1 Dataset Information
Our dataset `(AviationData.csv)` is historical aviation accident records, spanning several decades, from 1923 to 2022; and covering varying incidents from the United States (93%) and conditions. It contains `88863` different accident numbers; with `31` columns of info related to each. It also contains rows with missing data which may be as a result of differnt aircrafts having different error reporting methods. We also have a seperate small dataset of US state codes which will be useful in our location based analysis `(UseState_Codes.csv)`


#### 4.2 Dataset Summary

Our fields of concern that directly address our objectives and ultimately our problem statement include but are not limited to:


| **Field**             | **Description**                                           | **Data Type**  | **Relevance**                                             |
|-----------------------|-----------------------------------------------------------|----------------|-----------------------------------------------------------|
| Event.Id              | Unique accident identifier                                | String         | Tracks individual accident records                        |
| Event.Date            | Date of incident                                          | Date           | Enables time based trend and seasonal analysis           |
| Aircraft.Model        | Aircraft model name                                       | String         | Allows comparison of safety records among models           |
| Aircraft.damage       | Degree of aircraft damage             | Categorical    | Indicator of accident severity                |
| Number.of.Engines     | Number of engines on the aircraft                         | Integer        | Impact of this on safety       |
| Engine.Type           | Type of engine (Reciprocating, Turbo Fan)           | String         | Assists in identifying safe engine designs                  |
| Weather.Condition     | Weather conditions during the incident                    | String         | Assists in identifying performance under adverse conditions        |
| Broad.phase.of.flight | Phase of flight (takeoff, cruise, landing)          | String         | Identifies high-risk flight phases                          |
| Injury.Severity       | Level of injuries sustained (Fatal, Non-Fatal)      | String         | Risk analysis based on accident outcomes       |

*Source: [Kaggle](https://www.kaggle.com/datasets/khsamaha/aviation-accident-database-synopses/data)*

### 5. Data Exploration

Initial exploration to better understand dataset structure

#### 5.1 Importing libraries and data

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


#load data
df= pd.read_csv("./data/AviationData.csv", encoding='latin1')
states_df = pd.read_csv('./data/USState_Codes.csv', encoding='latin1')

C:\Users\Lewis\AppData\Local\Temp\ipykernel_25640\1704516843.py:8: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  df= pd.read_csv("./data/AviationData.csv", encoding='latin1')


#### 5.2 General overview of data

In [2]:
df.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude  Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2    Saltville, VA  United States  36.922223 -81.878056          NaN   
3       EUREKA, CA  United States        NaN        NaN          NaN   
4       Canton, OH  United States        NaN        NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [3]:
df.tail()

Event.Id Investigation.Type Accident.Number  Event.Date  \
88884  20221227106491           Accident      ERA23LA093  2022-12-26   
88885  20221227106494           Accident      ERA23LA095  2022-12-26   
88886  20221227106497           Accident      WPR23LA075  2022-12-26   
88887  20221227106498           Accident      WPR23LA076  2022-12-26   
88888  20221230106513           Accident      ERA23LA097  2022-12-29   

            Location        Country Latitude Longitude Airport.Code  \
88884  Annapolis, MD  United States      NaN       NaN          NaN   
88885    Hampton, NH  United States      NaN       NaN          NaN   
88886     Payson, AZ  United States  341525N  1112021W          PAN   
88887     Morgan, UT  United States      NaN       NaN          NaN   
88888     Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
88884          NaN  ...          Personal                 NaN   
88885          NaN  ...               NaN                 NaN   
88886       PAYSON  ...          Personal                 NaN   
88887          NaN  ...          Personal  MC CESSNA 210N LLC   
88888          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
88884                  0.0                    1.0                  0.0   
88885                  0.0                    0.0                  0.0   
88886                  0.0                    0.0                  0.0   
88887                  0.0                    0.0                  0.0   
88888                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight Report.Status  \
88884             0.0               NaN                    NaN           NaN   
88885             0.0               NaN                    NaN           NaN   
88886             1.0               VMC                    NaN           NaN   
88887             0.0               NaN                    NaN           NaN   
88888             1.0               NaN                    NaN           NaN   

      Publication.Date  
88884       29-12-2022  
88885              NaN  
88886       27-12-2022  
88887              NaN  
88888       30-12-2022  

[5 rows x 31 columns]

Succcesfuly loaded our data set. And we can immediately see the 31 columns we have and alot of missing values

In [4]:
df.describe()

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

- Fatal injuries statistics from `Total.Fatal.Injuries` reveals:
    - Mean of ~0.45 fatalities per incident
    - Maximum of 583 fatalities in a single incident
    - 75% of accidents had 0 fatalities (75th percentile is 0)

- Non fatal statistics from `Total.Serious.Injuries`, `Total.Minor.Injuries` reveal, and `Total.Uninjured`:
    - Average of ~0.14 serious injuries per incident
    - Mean of ~0.18 minor injuries
    - Mean of 1.64 uninjured persons tells us most accidents had survivors

- Aircrafts:
    - Most aircraft involved had 1-2 engines
    - Maximum number of engines in any incident was 8

This initial overview suggests that while aviation accidents are serious events, the majority result in no fatalities, with most people surviving the incidents. This aligns with general aviation safety sentiment on their safety

#### 5.2 Data shape, data types and missing data 

In [5]:
print(df.isnull().sum())

Event.Id                      0
Investigation.Type            0
Accident.Number               0
Event.Date                    0
Location                     52
Country                     226
Latitude                  54507
Longitude                 54516
Airport.Code              38757
Airport.Name              36185
Injury.Severity            1000
Aircraft.damage            3194
Aircraft.Category         56602
Registration.Number        1382
Make                         63
Model                        92
Amateur.Built               102
Number.of.Engines          6084
Engine.Type                7096
FAR.Description           56866
Schedule                  76307
Purpose.of.flight          6192
Air.carrier               72241
Total.Fatal.Injuries      11401
Total.Serious.Injuries    12510
Total.Minor.Injuries      11933
Total.Uninjured            5912
Weather.Condition          4492
Broad.phase.of.flight     27165
Report.Status              6384
Publication.Date          13771
dtype: i

In [6]:
print(f"Aviation data shape: {df.shape}")
print(f"State codes data shape: {states_df.shape}")

Aviation data shape: (88889, 31)
State codes data shape: (62, 2)


- Our aviation dataset has a total of `88889 rows` and `31 columns`. 
- Our states dataset is much smaller `62 rows` and `2 columns` will serve as a reference table for mapping state abbreviations to full names.

In [7]:
#check for data types and null values
df.isnull().sum()

Event.Id                      0
Investigation.Type            0
Accident.Number               0
Event.Date                    0
Location                     52
Country                     226
Latitude                  54507
Longitude                 54516
Airport.Code              38757
Airport.Name              36185
Injury.Severity            1000
Aircraft.damage            3194
Aircraft.Category         56602
Registration.Number        1382
Make                         63
Model                        92
Amateur.Built               102
Number.of.Engines          6084
Engine.Type                7096
FAR.Description           56866
Schedule                  76307
Purpose.of.flight          6192
Air.carrier               72241
Total.Fatal.Injuries      11401
Total.Serious.Injuries    12510
Total.Minor.Injuries      11933
Total.Uninjured            5912
Weather.Condition          4492
Broad.phase.of.flight     27165
Report.Status              6384
Publication.Date          13771
dtype: i

In [8]:
print("Unique values in Aircraft.damage:")
print(df['Aircraft.damage'].unique())
print("\nUnique values in Weather.Condition:")
print(df['Weather.Condition'].unique())
print("\nUnique values in Engine.Type:")
print(df['Engine.Type'].unique())
print("\nUnique values in Broad.phase.of.flight:")
print(df['Broad.phase.of.flight'].unique())
print("\nUnique values in Purpose.of.flight:")
print(df['Purpose.of.flight'].unique())
print("\nUnique values in Air.carrier:")
print(df['Air.carrier'].unique())


Unique values in Aircraft.damage:
['Destroyed' 'Substantial' 'Minor' nan 'Unknown']

Unique values in Weather.Condition:
['UNK' 'IMC' 'VMC' nan 'Unk']

Unique values in Engine.Type:
['Reciprocating' nan 'Turbo Fan' 'Turbo Shaft' 'Unknown' 'Turbo Prop'
 'Turbo Jet' 'Electric' 'Hybrid Rocket' 'Geared Turbofan' 'LR' 'NONE'
 'UNK']

Unique values in Broad.phase.of.flight:
['Cruise' 'Unknown' 'Approach' 'Climb' 'Takeoff' 'Landing' 'Taxi'
 'Descent' 'Maneuvering' 'Standing' 'Go-around' 'Other' nan]

Unique values in Purpose.of.flight:
['Personal' nan 'Business' 'Instructional' 'Unknown' 'Ferry'
 'Executive/corporate' 'Aerial Observation' 'Aerial Application'
 'Public Aircraft' 'Skydiving' 'Other Work Use' 'Positioning'
 'Flight Test' 'Air Race/show' 'Air Drop' 'Public Aircraft - Federal'
 'Glider Tow' 'Public Aircraft - Local' 'External Load'
 'Public Aircraft - State' 'Banner Tow' 'Firefighting' 'Air Race show'
 'PUBS' 'ASHO' 'PUBL']

Unique values in Air.carrier:
[nan 'Air Canada' 'Rocky M

 - Only a few columns don't have null values; we'll need to handle missing values for our relevant columns
 - Here we learn about the kind of categorizations we can do based on unique values in each relevant columns
 - Next Step: **Data Cleaning**


### 6. Data Cleaning

Here we'll be handling missing values, extreme ouliers, logically inconsistent data, and trying to normalize our data in every way possible to prepare data for accurate insights. 

#### 6.1 Handle missing values in focus areas


Let's handle missing values in our key fields, starting with the most critical ones for our analysis. We'll follow a systematic approach to maintain data integrity while maximizing usable data.

Choosing which columns to drop entirely based on their percentage of missing data and their usefulness in our analysis/

In [9]:
missing_percentage = (df.isnull().sum() / len(df) * 100).sort_values(ascending=False)
print("\nMissing Values %age:\n")
print(missing_percentage)

#drop all columns wth more than 50% missing values
df = df.loc[:, missing_percentage[missing_percentage < 50].index]

#drop columns of no interest and high missing value %age
df.drop(columns=['Airport.Code', 'Airport.Name','Publication.Date', 'Registration.Number', 'Amateur.Built', 'Broad.phase.of.flight'], inplace=True)
print(f"\n\nData shape after dropping columns: {df.shape}")


Missing Values %age:

Schedule                  85.845268
Air.carrier               81.271023
FAR.Description           63.974170
Aircraft.Category         63.677170
Longitude                 61.330423
Latitude                  61.320298
Airport.Code              43.601570
Airport.Name              40.708074
Broad.phase.of.flight     30.560587
Publication.Date          15.492356
Total.Serious.Injuries    14.073732
Total.Minor.Injuries      13.424608
Total.Fatal.Injuries      12.826109
Engine.Type                7.982990
Report.Status              7.181991
Purpose.of.flight          6.965991
Number.of.Engines          6.844491
Total.Uninjured            6.650992
Weather.Condition          5.053494
Aircraft.damage            3.593246
Registration.Number        1.554748
Injury.Severity            1.124999
Country                    0.254250
Amateur.Built              0.114750
Model                      0.103500
Make                       0.070875
Location                   0.058500
Inves

 - Dropped 9 columns bringing our currenty useful columns to 22; no rows dropped yet

Since aircraft identification is crucial, we'll drop rows with missing `Make` or `Model` values, which were a negligible percentage of `0.103500` and `0.070875` respectively, so we don't expect a huge loss of incidences. We'll also drop missing and unknown damage 

In [10]:
#length before dropping any rows
df_length = len(df)

df.dropna(subset=['Make', 'Model', 'Weather.Condition','Aircraft.damage', 'Injury.Severity'], inplace=True)
print(f"Records after dropping missing records: {df.shape}")
print(f"Dropped {df_length - len(df)} records")

Records after dropping missing records: (82254, 19)
Dropped 6635 records


In [11]:
#assume data was missing and never reported
injury_columns = ['Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Fatal.Injuries', 'Total.Uninjured']
df = df.dropna(subset=injury_columns, how='all')

#drop if total.people doesn't make sense
df[injury_columns] = df[injury_columns].fillna(0).astype(int)


For important columns with lots of missing values, we will fill them with relevant statistical values

In [12]:

columns_to_fill = ['Country', 'Purpose.of.flight', 'Location', 'Report.Status', 'Number.of.Engines', 'Engine.Type']

for col in columns_to_fill:
        if df[col].isnull().sum() > 0:
            mode_val = df[col].mode()[0]
            missing_count = df[col].isnull().sum()
            df.fillna({col: mode_val}, inplace=True)
            print(f"Filled {missing_count} missing values in {col} with '{mode_val}'")
    
print("\nAfter handling missing values:")
print(df[columns_to_fill].isnull().sum())

Filled 207 missing values in Country with 'United States'
Filled 2247 missing values in Purpose.of.flight with 'Personal'
Filled 43 missing values in Location with 'ANCHORAGE, AK'
Filled 2547 missing values in Report.Status with 'Probable Cause'
Filled 2791 missing values in Number.of.Engines with '1.0'
Filled 3223 missing values in Engine.Type with 'Reciprocating'

After handling missing values:
Country              0
Purpose.of.flight    0
Location             0
Report.Status        0
Number.of.Engines    0
Engine.Type          0
dtype: int64


In [13]:
print(f"Records after dropping missing records: {df.shape}")

Records after dropping missing records: (82176, 19)


After handling missing values:
- Dropped columns with a high missing %age and not useful to our analysis
- Dropped rows with missing data as they're essential for aircraft-risk value analysis
- Filled missing rows with mode values
- Current dataset has `82302` rows and `20` columns

In [14]:
df.isnull().sum()

Total.Serious.Injuries    0
Total.Minor.Injuries      0
Total.Fatal.Injuries      0
Engine.Type               0
Report.Status             0
Purpose.of.flight         0
Number.of.Engines         0
Total.Uninjured           0
Weather.Condition         0
Aircraft.damage           0
Injury.Severity           0
Country                   0
Model                     0
Make                      0
Location                  0
Investigation.Type        0
Event.Date                0
Accident.Number           0
Event.Id                  0
dtype: int64